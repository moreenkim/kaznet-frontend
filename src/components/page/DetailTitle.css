.detail-page-title {
  min-width: 150px;
  background-color: #f4f4f4;
}

.kaznet-detail-title {
  color: #464646;
  font-weight: bold;
  font-size: 2em;
  padding-top: 35px;
}

.kaznet-header-link:hover {
  text-decoration: none;
}

.kaznet-header-link {
  color: #711c27 !important;
}

.kaznet-badge {
  margin-left: 15px;
  font-weight: 100 !important;
  font-size: 20px !important;
  position: absolute;
}

.kaznet-creation-detail {
  opacity: 0.5;
}

.kaznet-action-links {
  color: #711c27 !important;
  padding-bottom: 10px;
  font-weight: bold;
}

.archive-button {
  color: #979797;
  position: absolute;
  right: 0px;
}

.archive-button:hover {
  text-decoration: none;
  color: #8c8c8c;
}

.action-link-divider {
  margin-left: 17px;
  margin-right: 17px;
}

.action-link,
.action-link.remove_button_css {
  font-weight: 600;
}

.action-link-alert,
.action-link.remove_button_css {
  color: #ac451e;
}

.withspace {
  margin-right: 10px;
}
