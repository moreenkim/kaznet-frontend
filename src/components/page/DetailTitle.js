// Renders the detail page title section
import React, { Component } from "react";
import FontAwesomeIcon from "@fortawesome/react-fontawesome";
import { Row, Col, Badge } from "reactstrap";
import { Link } from "react-router-dom";

export default class DetailPageTitle extends Component {
  render() {
    return (
      <section className="detail-page-title">
        <Col sm="12" md={{ size: 8, offset: 2 }}>
          <div>
            <h1 className="kaznet-detail-title">
              <Link to={this.props.pageTarget} className="kaznet-header-link">
                {this.props.pageTitle}
              </Link>{" "}
              > {this.props.detailName}
              <Badge className="kaznet-badge">Active</Badge>
            </h1>
            <p className="kaznet-creation-detail">
              By Jensen Nathan, 21 May 2018
            </p>
            <Col md="12">
              <Row className="kaznet-action-links">
                {/* Method maybe? to auto generate this links */}
                <Link to="/" className="action-link">
                  EDIT
                </Link>
                <Link to="/" className="action-link">
                  CREATE A COPY
                </Link>
                <Link to="/" className="action-link">
                  DEACTIVATE
                </Link>
                <Link to="/" className="action-link">
                  DELETE
                </Link>
              </Row>
            </Col>
          </div>
        </Col>
      </section>
    );
  }
}
