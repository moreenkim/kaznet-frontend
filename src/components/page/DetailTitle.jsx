// Renders the detail page title section
import React from 'react';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import { Row, Col, Badge } from 'reactstrap';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import './DetailTitle.scss';

const getStatusClassName = status => {
  const allowedStatuses = ['Active', 'Draft', 'Deactivated', 'Archived', 'Expired', 'Scheduled'];

  if (allowedStatuses.includes(status)) return `badge-${status.toLowerCase()}`;
  return '';
};

const DetailPageTitle = props => {
  const { actionLinks, detailName, status, pageTitle, pageTarget } = props;

  return (
    <section className="detail-page-title">
      <Col sm="12" md={{ size: 8, offset: 2 }}>
        <div>
          <h1 className="kaznet-detail-title">
            <Link to={pageTarget} className="kaznet-header-link">
              {pageTitle}
            </Link>
            &nbsp;&gt;&nbsp;
            {detailName}
            {status != null ? (
              <Badge className={`kaznet-badge ${getStatusClassName(status)}`}>{status}</Badge>
            ) : null}
          </h1>
          <p className="kaznet-creation-detail">By Jensen Nathan, 21 May 2018</p>
          <Col md="12">
            <Row className="kaznet-action-links">
              {/* Method maybe? to auto generate this links */}
              {actionLinks}
              <Link to="/" className="action-link archive-button">
                <FontAwesomeIcon icon="folder-open" className="withspace" />
                Archive
              </Link>
            </Row>
          </Col>
        </div>
      </Col>
    </section>
  );
};

DetailPageTitle.propTypes = {
  actionLinks: PropTypes.element.isRequired,
  detailName: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  pageTitle: PropTypes.string.isRequired,
  pageTarget: PropTypes.string.isRequired
};

export default DetailPageTitle;
