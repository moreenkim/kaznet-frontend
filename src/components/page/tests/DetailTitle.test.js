// test DetailTitle
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { Router } from "react-router";
import createBrowserHistory from "history/createBrowserHistory";

import DetailTitle from "../DetailTitle";

const history = createBrowserHistory();

describe("components/page/DetailTitle", () => {
  it("renders without crashing", () => {
    shallow(
      <DetailTitle
        pageTitle={"Kaznet"}
        detailName={"Testing!!!"}
        pageTarget={"/sol"}
        status={"Active"}
      />
    );
  });

  it("renders header correctly", () => {
    const wrapper = mount(
      <Router history={history}>
        <DetailTitle
          pageTitle={"Kaznet"}
          detailName={"Testing!!!"}
          pageTarget={"/sol"}
          status={"Active"}
        />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
