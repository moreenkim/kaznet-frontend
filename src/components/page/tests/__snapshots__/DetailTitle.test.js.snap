// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/page/DetailTitle renders header correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <DetailPageTitle
    detailName="Testing!!!"
    pageTarget="/sol"
    pageTitle="Kaznet"
    status="Active"
  >
    <section
      className="detail-page-title"
    >
      <Col
        md={
          Object {
            "offset": 2,
            "size": 8,
          }
        }
        sm="12"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="col-sm-12 col-md-8 offset-md-2"
        >
          <div>
            <h1
              className="kaznet-detail-title"
            >
              <Link
                className="kaznet-header-link"
                replace={false}
                to="/sol"
              >
                <a
                  className="kaznet-header-link"
                  href="/sol"
                  onClick={[Function]}
                >
                  Kaznet
                </a>
              </Link>
               
              &gt; 
              Testing!!!
              <Badge
                className="kaznet-badge badge-active"
                color="secondary"
                pill={false}
                tag="span"
              >
                <span
                  className="kaznet-badge badge-active badge badge-secondary"
                >
                  Active
                </span>
              </Badge>
            </h1>
            <p
              className="kaznet-creation-detail"
            >
              By Jensen Nathan, 21 May 2018
            </p>
            <Col
              md="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <div
                className="col-md-12"
              >
                <Row
                  className="kaznet-action-links"
                  tag="div"
                >
                  <div
                    className="kaznet-action-links row"
                  >
                    <Link
                      className="action-link archive-button"
                      replace={false}
                      to="/"
                    >
                      <a
                        className="action-link archive-button"
                        href="/"
                        onClick={[Function]}
                      >
                        <FontAwesomeIcon
                          border={false}
                          className="withspace"
                          fixedWidth={false}
                          flip={null}
                          icon="folder-open"
                          listItem={false}
                          mask={null}
                          name=""
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          symbol={false}
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-folder-open fa-w-18 withspace"
                            data-icon="folder-open"
                            data-prefix="fas"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 576 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M572.694 292.093L500.27 416.248A63.997 63.997 0 0 1 444.989 448H45.025c-18.523 0-30.064-20.093-20.731-36.093l72.424-124.155A64 64 0 0 1 152 256h399.964c18.523 0 30.064 20.093 20.73 36.093zM152 224h328v-48c0-26.51-21.49-48-48-48H272l-64-64H48C21.49 64 0 85.49 0 112v278.046l69.077-118.418C86.214 242.25 117.989 224 152 224z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                        Archive
                      </a>
                    </Link>
                  </div>
                </Row>
              </div>
            </Col>
          </div>
        </div>
      </Col>
    </section>
  </DetailPageTitle>
</Router>
`;
